src/test/integration/org/apache/cassandra/sidecar/routes/MultiDcTokenSupplier.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.routes;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.cassandra.distributed.api.TokenSupplier;
+
+/**
+ * Static factory holder that provides a multi-DC token supplier
+ */
+public class MultiDcTokenSupplier
+{
+
+    static TokenSupplier evenlyDistributedTokens(int numNodes, int numDcs, int numTokens)
+    {
+        long totalTokens = (long) numNodes * numDcs * numTokens;
+        BigInteger increment = BigInteger.valueOf((Long.MAX_VALUE / totalTokens) * 4);
```

Comment:
Where is `4` coming from?

Reply:
This was hard-coded to account for the no. DCs. Will update to be as folllows. `2` is as-is from the existing `TokenSupplier` (to account for the token-space being from Long.MIN - Long.MAX).

`(Long.MAX_VALUE / totalTokens)* 2 * numDCs`

We broaden the increment space by numDcs as the tokens assigned to neighboring nodes across different DCs are consecutive (i.e. delta 1).